packages/react/src/components/SelectUsersModal/SelectUsersModal.jsx

```diff
@@ -206,30 +217,32 @@ const SelectUsersModal = ({ isOpen, onClose, onSubmit, users, initialSelectedUse
     });
   };
 
-  const flattenUsers = useCallback((results, user) => {
-    return user.children && user.children.length > 0
-      ? user.children.reduce(flattenUsers, results)
-      : results.concat(user);
-  }, users);
-
   const usersList = displayAllUsersList(users, selectedUsers);
-  const userCount = usersList.reduce(flattenUsers, []).length;
+  const userCount = useMemo(() => usersList.reduce(flattenUsers, []).length, [usersList]);
   const selectedList = displaySelectedUsersList(selectedUsers);
 
   return (
     <div className={`${iotPrefix}--select-users-modal`}>
       <ComposedModal
         isLarge
         footer={{
-          isPrimaryButtonDisabled: selectedUsers.length === 0,
+          isPrimaryButtonDisabled: !canSaveRef.current,
+          primaryButtonLabel: i18n.primaryButtonLabel,
```

Comment:
Added primaryButtonLabel to footer, b/c the designs use an "OK" button instead of the default "Save" button used my the ComposedModal